### PR TITLE
Added timeoutafter to DotNetty eventloop calls

### PR DESCRIPTION
**Pull request by brotherneeru**
Updated dotnetty version to 0.7.5, as well as set a manual timeout for shutdown gracefully calls from dotnetty library that are hanging.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ x] Title of the pull request is clear and informative.
- [ x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MqttProtocolHead.cs [hunk 1]
```diff
@@ -107,17 +108,16 @@ public async Task CloseAsync(CancellationToken token)
...
                 this.logger.LogInformation("Stopping MQTT protocol head");
 
-                await (this.serverChannel?.CloseAsync() ?? TaskEx.Done);
-                await (this.eventLoopGroup?.ShutdownGracefullyAsync() ?? TaskEx.Done);
-                await (this.parentEventLoopGroup?.ShutdownGracefullyAsync() ?? TaskEx.Done);
-                await (this.wsEventLoopGroup?.ShutdownGracefullyAsync() ?? TaskEx.Done);
-                // TODO: gracefully shutdown the MultithreadEventLoopGroup in MqttWebSocketListener?
-                // TODO: this.webSocketListenerRegistry.TryUnregister("mqtts")?
+                await (this.serverChannel?.CloseAsync() ?? TaskEx.Done);
+                await TaskEx.TimeoutAfter(this.eventLoopGroup?.ShutdownGracefullyAsync(), TimeSpan.FromSeconds(TimeoutInSecs));
```
Comment (ancaantochi) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MqttProtocolHead.cs:112]: nit: if you add using: using Microsoft.Azure.Devices.Edge.Util;

you can then have: this.eventLoopGroup?.ShutdownGracefullyAsync()..TimeoutAfter(timeout);

also timeout can be declared as TimeSpan, like static readonly TimeSpan TimeoutInSecs = TimeSpan.FromSeconds(30);

**Review by ancaantochi — COMMENTED** [review]
During the meeting Varun also suggested to add a timeout at MainAsync level with 2 minutes, I think it would be good just to have additional timeout in case other protocols fail to stop. I think here would be the place to add it: https://github.com/Azure/iotedge/blob/main/edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs#L145

**Comment by brotherneeru** [comment]
@ancaantochi The logic in edgehub is different in 1.1:  https://github.com/Azure/iotedge/blob/65fffd366bbadf5599b8ce2689b0e87b51f2f447/edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs#L119

as such, I added the change in edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/EdgeHubProtocolHead.cs

**Review by ancaantochi — DISMISSED** [review]

**Review by ancaantochi — APPROVED** [review]

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/EdgeHubProtocolHead.cs [hunk 2]
```diff
@@ -31,7 +33,15 @@ public Task StartAsync()
         public Task CloseAsync(CancellationToken token)
         {
             this.logger.LogInformation($"Closing protocol heads - {this.Name}");
-            return Task.WhenAll(this.underlyingProtocolHeads.Select(protocolHead => protocolHead.CloseAsync(token)));
+            try
+            {
+                return Task.WhenAll(this.underlyingProtocolHeads.Select(protocolHead => protocolHead.CloseAsync(token))).TimeoutAfter(TimeoutInSecs);
```
Comment (ancaantochi) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/EdgeHubProtocolHead.cs:38]: should you use TaskEx.TimeoutAfter here too until we find out why task.TimeoutAfter doesn't work?

**Review by ancaantochi — APPROVED** [review]

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/EdgeHubProtocolHead.cs (hunk 2 above)
Comment (varunpuranik) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/EdgeHubProtocolHead.cs:38]: task.TimeoutAfter doesn't work?! Why?

**Review by ancaantochi — APPROVED** [review]

**Pull request merged**